scripts/update-component-sha.sh

```diff
@@ -1,10 +1,66 @@
 #!/bin/sh
 set -e
-if [ $# -ne 2 ] ; then
-    echo "Need <OLD> and <NEW> as arguments" >&2
+
+##
+#
+# script to replace hashes in config files
+# can be run in 2 modes:
+# 1- $0 --hash $oldhash $newhash              - replace all instances of $oldhash with $newhash
+# 2- $0 --image $image $newhash        - replace all hashes for $image with $newhash
+#
+# for backwards compatibility if no argtype is given, the first method is assumed. Thus the following are equivalent:
+#   $0 $oldhash $newhash 
+#   $0 --hash $oldhash $newhash 
+#
+
+function usage() {
+    echo "Modes of operation:" >&2
+    echo >&2
+    echo "Replace by hash:" >&2
+    echo "	$0 --hash <OLD> <NEW>" >&2
+    echo "	Example: $0 8675309abcdefg abcdef567899" >&2
```

Comment:
Sounds like we have been using it in different ways. I've (and @rn?) use it as:
```
./scripts/update-component-sha.sh linuxkitprojects/swarmd:1cd4c061cc7327750d2a12c267db6d4d9e26b1d3 linuxkitprojects/swarmd:cce587b5e67b7d3e088a2e527dafc5ad3edb6502
```
Whereas you seem to use it as
```
./scripts/update-component-sha.sh 1cd4c061cc7327750d2a12c267db6d4d9e26b1d3  cce587b5e67b7d3e088a2e527dafc5ad3edb6502
```

If we discount the possibility of hash clashes (super unlikely) then they are equivalent. I just the first form because I can easily pick it out from the `show-tag` output, or just from the build log.